paddle/fluid/framework/operator.cc

```diff
@@ -1102,10 +1106,9 @@ Scope* OperatorWithKernel::PrepareData(
   }
 
   for (auto& var_name_item : Inputs()) {
+    bool should_skip_input = false;
     if (no_buffer_ins && no_buffer_ins->count(var_name_item.first) > 0) {
-      VLOG(7) << "Skip scanning input " << var_name_item.first
-              << " in Operator " << type_;
-      continue;
+      should_skip_input = true;
```

Comment:
Wouldn't be here more clear to just initialize should_skip_input with value of predicate inside if()?

Reply:
Good catch! Will change that.